packages/camera/camera_avfoundation/example/ios/RunnerTests/AvailableCamerasTest.m

```diff
@@ -73,6 +73,77 @@ - (void)testAvailableCamerasShouldReturnAllCamerasOnMultiCameraIPhone {
+  OCMStub([wideAngleCamera uniqueID]).andReturn(@"0");
+  OCMStub([wideAngleCamera position]).andReturn(AVCaptureDevicePositionBack);
+
+  AVCaptureDevice *frontFacingCamera = OCMClassMock([AVCaptureDevice class]);
+  OCMStub([frontFacingCamera uniqueID]).andReturn(@"1");
+  OCMStub([frontFacingCamera position]).andReturn(AVCaptureDevicePositionFront);
+
+  AVCaptureDevice *ultraWideCamera = OCMClassMock([AVCaptureDevice class]);
+  OCMStub([ultraWideCamera uniqueID]).andReturn(@"2");
+  OCMStub([ultraWideCamera position]).andReturn(AVCaptureDevicePositionBack);
+
+  AVCaptureDevice *telephotoCamera = OCMClassMock([AVCaptureDevice class]);
+  OCMStub([telephotoCamera uniqueID]).andReturn(@"3");
+  OCMStub([telephotoCamera position]).andReturn(AVCaptureDevicePositionBack);
+
+  // iPad External Camera
+  AVCaptureDevice *externalCamera = OCMClassMock([AVCaptureDevice class]);
+  OCMStub([externalCamera uniqueID]).andReturn(@"4");
+  OCMStub([externalCamera position]).andReturn(AVCaptureDevicePositionUnspecified);
+
+  NSMutableArray *requiredTypes =
+      [@[ AVCaptureDeviceTypeBuiltInWideAngleCamera, AVCaptureDeviceTypeBuiltInTelephotoCamera ]
+          mutableCopy];
+  if (@available(iOS 13.0, *)) {
+    [requiredTypes addObject:AVCaptureDeviceTypeBuiltInUltraWideCamera];
+  }
+  if (@available(iOS 17.0, *)) {
+    [requiredTypes addObject:AVCaptureDeviceTypeExternal];
+  }
+  id discoverySessionMock = OCMClassMock([AVCaptureDeviceDiscoverySession class]);
+  OCMStub([discoverySessionMock discoverySessionWithDeviceTypes:requiredTypes
+                                                      mediaType:AVMediaTypeVideo
+                                                       position:AVCaptureDevicePositionUnspecified])
+      .andReturn(discoverySessionMock);
+
+  NSMutableArray *cameras = [NSMutableArray array];
+  [cameras addObjectsFromArray:@[ wideAngleCamera, frontFacingCamera, telephotoCamera ]];
+  if (@available(iOS 13.0, *)) {
+    [cameras addObject:ultraWideCamera];
+  }
+  if (@available(iOS 17.0, *)) {
+    [cameras addObject:externalCamera];
+  }
+  OCMStub([discoverySessionMock devices]).andReturn([NSArray arrayWithArray:cameras]);
+
+  MockFLTThreadSafeFlutterResult *resultObject =
+      [[MockFLTThreadSafeFlutterResult alloc] initWithExpectation:expectation];
+
+  // Set up method call
+  FlutterMethodCall *call = [FlutterMethodCall methodCallWithMethodName:@"availableCameras"
+                                                              arguments:nil];
+
+  [camera handleMethodCallAsync:call result:resultObject];
+
+  // Verify the result
+  NSDictionary *dictionaryResult = (NSDictionary *)resultObject.receivedResult;
+  if (@available(iOS 17.0, *)) {
+    XCTAssertTrue([dictionaryResult count] == 5);
+  } else if (@available(iOS 13.0, *)) {
+    XCTAssertTrue([dictionaryResult count] == 4);
+  } else {
+    XCTAssertTrue([dictionaryResult count] == 3);
```

Comment:
can you verify the actual camera IDs in the result, rather than just the count?

Reply:
i may be missing something. why was this resolved?

Reply:
Ah I just noticed this was for the other test. This was a pre-existing test not in the scope of this PR. Maybe another PR should be openned to write better tests for available cameras

Reply:
hmmm, it looks like it's still checking `dictionaryResult`'s count only. can you point me to the code?

Reply:
![image](https://github.com/flutter/packages/assets/67685395/0f7b265d-a54d-47c8-84aa-10defb1f62cf)

I think you may be looking at an older commit the latest one looks like the above

Reply:
@hellohuanlin Just wanted to make sure my previous message wasn't lost in your inbox

Reply:
instead of checking the `[dictionaryResult count] == 3`, can you manually check the actual items?

Reply:
Looks like this is the last outstanding comment before this can be merged?

Reply:
Yes I will work on it today and update the test